### PR TITLE
Close #11877: Add support to sanitize intent

### LINKED ISSUE #11877: Sanitize external intents
We don't currently sanitize intents correctly, see https://bugzilla.mozilla.org/show_bug.cgi?id=1759760#c3

### PR DESCRIPTION
**Pull request by rocketsroger**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/feature/intent/src/test/java/mozilla/components/feature/intent/ext/IntentExtensionsTest.kt [hunk 1]
```diff
@@ -40,4 +43,41 @@ class IntentExtensionsTest {
...
+    fun `WHEN sanitize throw no exception THEN no extra is removed`() {
+        val intent: Intent = mock()
+        `when`(intent.getBooleanExtra("TriggerUnparcel", false)).thenReturn(false)
+
+        intent.sanitize()
+        verify(intent, never()).replaceExtras(null)
+    }
+
+    @Test
+    fun `WHEN get extra throws BadParcelableException THEN clear all extras`() {
```
Comment (csadilek) [inline components/feature/intent/src/test/java/mozilla/components/feature/intent/ext/IntentExtensionsTest.kt:57]: nit: WHEN unparcel fails with BadParcelableException THEN extras are cleared.

File: components/feature/intent/src/test/java/mozilla/components/feature/intent/ext/IntentExtensionsTest.kt [hunk 2]
```diff
@@ -40,4 +43,41 @@ class IntentExtensionsTest {
...
+        val intent: Intent = mock()
+        `when`(intent.getBooleanExtra("TriggerUnparcel", false)).thenThrow(BadParcelableException("test"))
+        `when`(intent.replaceExtras(null)).thenReturn(intent)
+
+        intent.sanitize()
+        verify(intent).replaceExtras(null)
+    }
+
+    @Test
+    fun `WHEN get extra throws RuntimeException with ClassNotFoundException cause THEN clear all extras`() {
```
Comment (csadilek) [inline components/feature/intent/src/test/java/mozilla/components/feature/intent/ext/IntentExtensionsTest.kt:67]: nit: naming see above.

File: components/feature/intent/src/test/java/mozilla/components/feature/intent/ext/IntentExtensionsTest.kt [hunk 3]
```diff
@@ -40,4 +43,41 @@ class IntentExtensionsTest {
...
+        val intent: Intent = mock()
+        `when`(intent.getBooleanExtra("TriggerUnparcel", false)).thenThrow(RuntimeException("test", ClassNotFoundException("test")))
+        `when`(intent.replaceExtras(null)).thenReturn(intent)
+
+        intent.sanitize()
+        verify(intent).replaceExtras(null)
+    }
+
+    @Test(expected = RuntimeException::class)
+    fun `WHEN get extra throws RuntimeException THEN throws exception`() {
```
Comment (csadilek) [inline components/feature/intent/src/test/java/mozilla/components/feature/intent/ext/IntentExtensionsTest.kt:77]: nit: naming see above.

**Review by csadilek — APPROVED** [review]
Some nits below, otherwise 🚢

File: components/feature/intent/src/test/java/mozilla/components/feature/intent/ext/IntentExtensionsTest.kt [hunk 4]
```diff
@@ -40,4 +43,41 @@ class IntentExtensionsTest {
         assertEquals(id, intent.getSessionId())
         assertEquals(id, intent.toSafeIntent().getSessionId())
     }
+
+    @Test
+    fun `WHEN sanitize throw no exception THEN no extra is removed`() {
```
[the remark this answers is not in the document]
Comment (rocketsroger) [reply components/feature/intent/src/test/java/mozilla/components/feature/intent/ext/IntentExtensionsTest.kt:48]: Sure, I'll update test names.

**Pull request merged**